store/postgres/src/store.rs

```diff
@@ -712,9 +715,26 @@ impl Store {
         Ok(storage.clone())
     }
 
-    fn subgraph_info(&self, subgraph_id: &SubgraphDeploymentId) -> Result<SubgraphInfo, Error> {
-        if let Some(info) = self.subgraph_cache.lock().unwrap().get(&subgraph_id) {
-            return Ok(info.clone());
+    fn cached_schema(&self, subgraph_id: &SubgraphDeploymentId) -> Result<SchemaCacheEntry, Error> {
+        if let Some(entry) = self.schema_cache.lock().unwrap().get(&subgraph_id) {
+            // Cache entry is stale and an unstable import has changed or is now available
+            // TODO: Is it okay to ignore unstable imports which are no longer available?
+            // TODO: Make this threshold an envvar?
+            let requires_refresh = entry.last_refresh.elapsed().as_secs() >= 120
+                && entry
+                    .unstable_imports
+                    .iter()
+                    .any(|(schema_reference, subgraph_id_opt)| {
+                        self.resolve_schema_reference(schema_reference)
+                            .map(|schema| match subgraph_id_opt {
+                                Some(subgraph_id) => schema.id.eq(subgraph_id),
+                                None => true,
+                            })
+                            .unwrap_or(false)
+                    });
```

Comment:
The body of this `any` took me quite a bit to understand. I think what it is saying is "we need to refresh this entry if (a) we have an unstable import that we couldn't resolve last time around or (b) is an unstable import points at a different subgraph; we ignore errors in resolving a reference in all cases" It would be a bit clearer if the `match subgraph_id_opt` happened before the call to `resolve_schema_reference`